src/plugins/legacy/polygonRoi/polygonEventFilter.cpp

```diff
@@ -68,145 +69,137 @@ void polygonEventFilter::updateView(medAbstractImageView *view)
-        QMenu *roiManagementMenu = new QMenu("Delete");
-        if (closestManager && closestManager->getMinimumDistanceFromNodesToEventPosition(mousePos) < 10 )
-        {
-            mainMenu.addMenu(roiManagementMenu);
-            QAction *deleteOneNodeAction = new QAction("One node", roiManagementMenu);
-            connect(deleteOneNodeAction, &QAction::triggered, [=](){
-                emit nodeToDelete(closestManager, mousePos[0], mousePos[1]);
-            });
-            roiManagementMenu->addAction(deleteOneNodeAction);
-
-            QAction *deleteContour = new QAction("Contour", roiManagementMenu);
-            connect(deleteContour, &QAction::triggered, [=](){
-                emit contourToDelete(closestManager);
-            });
-            roiManagementMenu->addAction(deleteContour);
-
-            QAction *deleteLabel = new QAction("Label", roiManagementMenu);
-            connect(deleteLabel, &QAction::triggered, [=](){
-                emit labelToDelete(closestManager);
-            });
-            roiManagementMenu->addAction(deleteLabel);
-
-            QAction *saveMask = new QAction("Save Mask", &mainMenu);
-            connect(saveMask, &QAction::triggered, [=](){
-                emit maskToSave(closestManager);
-            });
-
-            mainMenu.addMenu(roiManagementMenu);
-            mainMenu.addAction(saveMask);
-        }
-        else
-        {
-            labelMenu->setTitle("Label");
-            mainMenu.addMenu(labelMenu);
-        }
+    return false;
+}
 
+bool polygonEventFilter::mousePressEvent(medAbstractView * view, QMouseEvent *mouseEvent)
+{
+    if ( ! activateEventFilter )
+        return false;
 
-        mainMenu.exec(mouseEvent->globalPos());
-        delete labelMenu;
-        delete roiManagementMenu;
+    medAbstractImageView *v = dynamic_cast<medAbstractImageView *> (view);
+    if ( !currentView )
+    {
+        return false;
+    }
+    if (currentView != v)
+        return false;
 
-        return true;
+    if (mouseEvent->button()==Qt::LeftButton)
+    {
+        return leftButtonBehaviour(view, mouseEvent);
+    }
+    else if (mouseEvent->button()==Qt::RightButton)
+    {
+        return rightButtonBehaviour(view, mouseEvent);
     }
     return false;
 }
 
+bool polygonEventFilter::updateMainViewOnChosenSlice(medAbstractImageView *view, QMouseEvent *mouseEvent)
+{
+    QPointF position;
+    position.setX(mouseEvent->x()*QGuiApplication::screenAt(mouseEvent->pos())->devicePixelRatio());
+    position.setY(mouseEvent->y()*QGuiApplication::screenAt(mouseEvent->pos())->devicePixelRatio());
+    QVector3D clickPosition = view->mapDisplayToWorldCoordinates(position);
+    int slice = findClosestSliceFromMouseClick(clickPosition);
+    if ( slice == -1 )
+        return false;
+    vtkImageView2D *view2D =  static_cast<medVtkViewBackend*>(currentView->backend())->view2D;
+    view2D->SetSlice(slice);
+    view2D->Render();
+    return false;
```

Comment:
the default behavior is `false`?

Reply:
I had some bugs when I tried to return true...
Not sure to understand well how it works... But it works...

Reply:
this function is called in `mouseReleaseEvent`. Is it possible that these functions are not using true and false, but 0 and 1 as success and failure in C? If yes, it would be important to look at that, because for instance, line 186 you would exit the function with 0, a success.